model/relation/event/processor/ItemRemovedEventProcessor.php

```diff
@@ -50,17 +50,30 @@ public function process(Event $event): void
         }
 
         if ($deleteRelatedAssets) {
-            $collection = $this->getMediaRelationRepository()->getItemAssetUris($id);
+            $assetsToDelete = [];
 
-            $this->getAssetDeleter()->deleteAssetsByURIs($collection);
+            $mediaRelationRepository = $this->getMediaRelationRepository();
+            foreach ($mediaRelationRepository->getItemAssetUris($id) as $assetUri) {
+                $relatedItemUris = array_unique(
+                    $mediaRelationRepository->getRelatedItemUrisByAssetUri($assetUri)
+                );
 
-            $this->getLogger()->info(
-                sprintf(
-                    'Assets "%s" removed after Item "%s" using them was removed ',
-                    json_encode($collection),
-                    $id
-                )
-            );
+                if ((count($relatedItemUris) === 1) && (current($relatedItemUris) === $id)) {
```

Comment:
You do not need to compare the ID again, it was part of your query already. Lets watch out the overvalidation.

```suggestion
                if (count($relatedItemUris) === 1) {
```

Reply:
Removed in 59b249b93086b877e286a84d4925e856f08b36f9.